kolibri/core/content/utils/channels.py

```diff
@@ -30,18 +33,29 @@ def get_channel_ids_for_content_database_dir(content_database_dir):
         logger.warning("Ignoring databases in content database directory '{directory}' with invalid names: {names}"
                        .format(directory=content_database_dir, names=invalid_db_names))
 
+    # nonexistent database files are created if we delete the files that have broken symbolic links;
     # empty database files are created if we delete a database file while the server is running and connected to it;
     # here, we delete and exclude such databases to avoid errors when we try to connect to them
-    empty_db_files = set({})
+    db_files_to_remove = set({})
     for db_name in valid_db_names:
         filename = os.path.join(content_database_dir, "{}.sqlite3".format(db_name))
-        if os.path.getsize(filename) == 0:
-            empty_db_files.add(db_name)
+        if not os.path.exists(filename) or os.path.getsize(filename) == 0:
+            db_files_to_remove.add(db_name)
             os.remove(filename)
-    if empty_db_files:
-        logger.warning("Removing empty databases in content database directory '{directory}' with IDs: {names}"
-                       .format(directory=content_database_dir, names=empty_db_files))
-    valid_dbs = list(set(valid_db_names) - set(empty_db_files))
+            # Delete the channel metadata from the database if exists, so that
+            # users can download the database file again from the channel import page.
+            channel = ChannelMetadata.objects.filter(id=db_name)
```

Comment:
yes that makes sense. I'm doing this here because if the user tries to export the channel with a corrupted DB file to USB, then a FileNotFound error will be raised and the user will not be able to export the channel unless they manually download the DB file from studio and put that into Kolibri home directory. Deleting the channel from the default database will not remove the existing content of the channel. If the user go to the channel import page, they will get to download the channel DB file and still see the existing content. However, it will definitely be confusing to them that the channel disappears with no indication. Do you have any suggestions on handling this case? Thank you!

Reply:
From the user's perspective, deleting the channel from the default database does _delete_ the channel content (because they will not be able to see it in the interface any more).

So, I think it would be better to catch the FileNotFound error in the export step, so that only one user flow is disrupted (export) and not two (interacting with content and export).

Reply:
yes that sounds better! I assume that we can just throw the FileNotFound exception in the export step without doing any extra implementations?